app/views/carts/show.html.erb

```diff
@@ -18,7 +18,13 @@
     <tbody>
     <% @cart.line_items.each do |line| %>
       <tr>
-        <th scope="row"><%= image_tag(line.item.image_url, size: "100x100") %></th>
+        <th scope="row">
```

Comment:
Le commentaire suivant sera valable pour toutes les situations identiques ci-dessous.

Si je comprends bien tu veux t'épargner le bug "no method variant to nil class" quand ton image est vide.

Mais tu n'affiches rien si elle est vide en dessous (par exemple afficher une image par défaut extraite du fichier asset).

Dans ce cas là, écris le ainsi 👍 

<%= image_tag(line.item.picture.variant(resize("100x100"), alt:"picture") if line.item.picture.attached? %>

Tu n'as pas besoin de tester l'url. C'est long et pas très lisible. Tu veux juste savoir si ton objet a une image attachée.

Très bien, Active storage a une méthode : attached?

Quand tu l'appliques à ton objet, elle renvoie true si un objet est attaché. Bcp plus rapide et bien plus lisible.

Reply:
En fait, je cherche à savoir si mon objet a une image_url car mes premiers items crées via le seed on une image_url et mes mes items crées via le CRUD on une image attached.
A terme ton les items auront une image attached et je changerai le code.